### DISCUSSION TITLE
Deconvolution of large polymer (~2000 Da) spectrum

### DISCUSSION THREAD

**Discussion by mengxuanjia** [mobiusklein/ms_deisotope#22] [General]
Hi,

I'm working with polymer ESI spectrum using ms_deisotope. I find for ions around 2000 Da (doubly charged) the deconvolution does not work well due to the isotope. I played with the tuncate_after and incremental_truncation but was not able to find a good combination. Can you offer any suggestions? Thank you!

Mengxuan

Here is an example spectrum:

m/z | Intensity
512.7523 | 3701.523
522.2719 | 24233.62
528.3567 | 2343.17
546.4907 | 29049.13
547.4938 | 10258.28
599.5426 | 20841.17
610.1875 | 25780.57
611.1882 | 13600.56
612.1857 | 9131.48
613.1888 | 1164.305
678.7798 | 7686.948
680.4848 | 22387.27
681.487 | 8917.356
682.5314 | 27738.6
683.5329 | 10315.05
695.3306 | 20499.6
698.5494 | 4494.993
701.6206 | 4440.987
702.6628 | 2095.218
708.5166 | 14618.18
709.5175 | 5367.887
738.5334 | 24083.79
739.5354 | 7564.605
816.0481 | 22648.18
847.9105 | 2657.917
857.4657 | 5336.473
858.4915 | 1438.397
869.8629 | 10399.25
870.6154 | 1115.025
891.9035 | 14368.85
892.5916 | 4963.443
893.2284 | 1313.944
913.6554 | 10205.6
914.3632 | 16298.39
915.2803 | 2594.064
935.8598 | 18617.74
936.7072 | 9589.289
937.2614 | 7233.339
938.3249 | 1146.908
944.6689 | 1167.562
957.8045 | 11328.69
958.6482 | 24180.63
959.2762 | 7716.723
959.8263 | 3739.728
966.8287 | 1189.08
971.1602 | 25527.57
979.8139 | 8571.736
980.5729 | 19678.74
981.1368 | 15990.23
981.7953 | 8856.333
1000.689 | 1192.681
1001.801 | 11983.92
1002.642 | 25355.44
1003.408 | 30995.84
1004.208 | 2075.912
1023.768 | 8274.974
1024.666 | 28119.84
1025.386 | 36868.04
1026.304 | 6418.654
1044.769 | 1526.955
1045.832 | 7758.878
1046.67 | 28469.52
1047.244 | 28806.01
1047.901 | 19914.15
1060.671 | 4012.074
1067.739 | 1049.714
1068.731 | 25413.6
1069.41 | 44296.56
1070.359 | 6557.457
1088.871 | 2204.357
1089.978 | 7111.205
1091.093 | 56351.6
1091.947 | 17772.74
1104.934 | 1952.599
1111.926 | 1055.242
1112.764 | 20895.07
1113.448 | 37595.58
1114.274 | 5892.544
1114.774 | 1248.593
1132.842 | 1475.549
1134.8 | 17353.85
1135.365 | 22975.86
1135.953 | 12548.68
1136.536 | 1633.067
1143.47 | 1198.823
1153.697 | 4931.489
1157.067 | 26810.67
1157.809 | 9172.497
1158.32 | 3984.268
1175.106 | 28400.03
1176.863 | 1494.988
1178.839 | 10495.86
1179.481 | 15389.85
1180.315 | 1156.287
1183.236 | 4218.603
1186.93 | 1078.75
1188.884 | 10974.69
1195.524 | 1105.054
1200.793 | 4571.903
1201.501 | 9240.234
1202.13 | 2314.735
1212.879 | 1355.626
1220.45 | 1943.427
1223.113 | 7618.055
1223.973 | 5270.047
1225.152 | 1600.052
1252.497 | 1591.737
1274.496 | 2079.51
1318.129 | 2462.013
1321.13 | 1643.324
1340.619 | 1753.06
1361.824 | 2856.169
1362.569 | 4561.272
1364.224 | 1103.785
1372.178 | 3549.526
1379.289 | 1733.072
1386.555 | 1407.333
1408.666 | 1027.255
1428.35 | 1503.285
1434.684 | 2534.404
1439.025 | 1682.444
1443.195 | 14330.02
1450.755 | 44909.67
1469.62 | 5383.178
1471.677 | 2397.051
1474.489 | 1455.233
1484.511 | 39440.19
1486.482 | 1432.059
1490.561 | 1516.097
1510.316 | 3693.358
1515.828 | 1305.183
1516.638 | 1037.095
1521.019 | 2654.813
1525.572 | 1365.057
1526.355 | 4822.771
1529.816 | 1079.276
1530.614 | 1089.573
1542.508 | 1385.961
1553.48 | 1140.581
1575.054 | 1155.653
1577.489 | 3377.727
1589.118 | 1723.625
1590.414 | 1181.364
1591.316 | 1992.393
1598.518 | 6981.395
1602.448 | 1419.249
1603.415 | 1999.851
1615.399 | 1124.169
1649.487 | 5777.415
1656.934 | 11128.92
1658.449 | 4287.376
1664.589 | 2139.96
1665.969 | 2129.92
1672.029 | 2760.828
1673.503 | 1216.586
1676.957 | 1913.872
1678.254 | 8013.96
1679.622 | 14598.7
1682.937 | 4558.233
1685.831 | 7388.806
1687.176 | 1214.241
1690.553 | 5177.834
1710.713 | 7862.6
1718.607 | 9937.737
1720.402 | 5426.299
1721.301 | 11282.35
1723.83 | 13838.78
1757.836 | 1483.66
1758.833 | 1304.521
1759.493 | 25051.68
1761.029 | 1863.644
1762.109 | 3007.404
1765.914 | 2663.329
1767.735 | 6587.699
1772.786 | 14377.98
1775.343 | 2184.022
1781.073 | 6477.935
1784.065 | 13541.35
1784.992 | 5055.288
1789.449 | 7283.795
1801.641 | 2339.187
1803.083 | 3279.198
1809.121 | 2766.897
1836.079 | 13581.35
1866.64 | 1299.295
1870.06 | 2287.573
1870.637 | 3339.272
1871.601 | 4130.627
1873.5 | 4337.97
1879.294 | 5775.514
1885.556 | 6528.386
1888.614 | 2697.283
1893.768 | 2008.466
1897.817 | 3140.604
1901.254 | 1045.935
1916.225 | 7351.067
1917.773 | 2326.486
1925.662 | 1927.806
1931.289 | 6872.995
1935.202 | 1942.59
1940.499 | 2645.188
1944.595 | 6287.598
1946.384 | 4786.199
1954.141 | 6528.873
1961.991 | 2335.988
1964.201 | 17222.54
1971.773 | 1439.62
1977.19 | 6849.213
1983.253 | 1276.306
1993.12 | 8607.993
1994.382 | 2564.557
1999.501 | 3318.944

**Comment by mobiusklein** [comment]
Is it failing to give the correct monoisotopic peak or to give any solutions at all? Are you specifying a specific score threshold?

The default isotopic model uses Senko's peptide averagine. This dictates how to generate isotopic patterns as a function of mass and charge. If your molecule type isn't amino acid-like, you might need a different averagine. Do you know the chemical composition of the average monomer of the material you're analyzing?

**Reply by mengxuanjia** [reply]
It fails to give any solutions for the polymers. I average all the identified molecules and got C59.97, H118.90 O19.96 N2. The repeating unit is C2H4O. If I use the number of repeating units to average it is C4.276 H8.4786 O1.4238 N 0.1426. I tested with the new averagine, and got similar results. The deconvolution works for most of my ions smaller than 1800 Da and could not work when the masses are higher.

**Reply by mobiusklein** [reply]
What kind of detector are you using? I see a lot of peak envelopes that are  *close* but not quite a neutron/charge apart at 20 PPM. 

```python
import io
import ms_deisotope
from ms_deisotope.data_source.text import scan_from_csv

buff = io.BytesIO('''<your spectrum text here>''')
scan = scan_from_csv(buff, "|", is_profile=False)
scan.pick_peaks()

mono = dict(C=4.276, H=8.4786, O=1.4238, N=0.1426)
model = ms_deisotope.Averagine(mono)

scan.deconvolute(averagine=model, scorer=ms_deisotope.PenalizedMSDeconVFitter(0), truncate_after=0.95,
                             # I had to really boost this up to get anything to return. I don't trust these fits.
                             error_tolerance=2e-4)
```

Then, using `scan.plot.annotate_isotopic_peaks()` to visualize the envelopes, I selected an area and saw this:
![image](https://user-images.githubusercontent.com/1426041/109363733-3f0b6300-785b-11eb-9127-c29dcb076190.png)
```python
In [70]: [p.mz for p in scan.peak_set.between(1134, 1137)]
Out[70]: [1134.8, 1135.365, 1135.953, 1136.536]
In [71]: np.diff([p.mz for p in scan.peak_set.between(1134, 1137)])
Out[71]: array([0.565, 0.588, 0.583])
```
For a 2+ ion, this should be close to 0.5016774189.

**Reply by mengxuanjia** [reply]
I used Orbi at 17k resolution. Thank you for point this out! I will play with the mass spec settings and check the results.

Many thanks!
